tests/PHPStan/Analyser/data/literal-string.php

```diff
@@ -34,6 +37,25 @@ public function doFoo($literalString, string $string)
 			str_repeat('a', 99)
 		);
 		assertType('literal-string&non-falsy-string', str_repeat('a', 100));
+		assertType('literal-string&non-empty-string&numeric-string', str_repeat('0', 100)); // could be non-falsy-string
+		assertType('literal-string&non-falsy-string&numeric-string', str_repeat('1', 100));
+		// Repeating a numeric type multiple times can lead to a non-numeric type: 3v4l.org/aRBdZ
+		assertType('non-empty-string', str_repeat($numericString, 100));
+
+		assertType("''", str_repeat('1.23', 0));
+		assertType("''", str_repeat($string, 0));
+		assertType("''", str_repeat($numericString, 0));
+
+		// see https://3v4l.org/U4bM2
+		assertType("non-empty-string", str_repeat($numericString, 1)); // could be numeric-string
+		assertType("non-empty-string", str_repeat($numericString, 2));
+		assertType("literal-string", str_repeat($literalString, 1));
+		$x = rand(1,2);
+		assertType("literal-string&non-falsy-string", str_repeat('    1   ', $x));
+		assertType("literal-string&non-falsy-string", str_repeat('+1', $x));
+		assertType("literal-string&non-falsy-string", str_repeat('1e9', $x));
+		assertType("literal-string&non-falsy-string&numeric-string", str_repeat('19', $x));
```

Comment:
good catch. I forgot to handle a maybe zero multiplyer.

the type as you suggested wouldn't work though, because while type normalization it gets less precise and will just lead to a `literal-string`.